microfs.py

```diff
@@ -61,37 +62,43 @@ def raw_on(serial):
     """
     Puts the device into raw mode.
     """
```

Comment:
I've refactored this function slightly to make the main raw-mode procedure easier to follow, but that has increased the diff. The only real change to this function is at the last step.

The original function ends with this after the soft-reset (checks that the raw-REPL message has been sent):

```python
data = serial.read_until(b'raw REPL; CTRL-B to exit\r\n>')
if not data.endswith(b'raw REPL; CTRL-B to exit\r\n>'):
    if COMMAND_LINE_FLAG:
        print(data)
    raise IOError('Could not enter raw REPL.')
```

Most MicroPython boards drop you back into the raw-REPL at this point, since that was the active mode before the soft reset.
However, Circuitpython has actually changed the message sent and **also** asks for any key to be sent before dropping you back into the raw-REPL.

```
PYB: soft reboot
#8 ets_task(401003b0, 3, 3fff83f0, 4)
boot.py output:

Press any key to enter the REPL. Use CTRL-D to soft reset.
```

So the main change here is that if the raw-repl message is not received after the soft-reset, it will send one last "Ctrl+A" command and check again:

```python
# Some MicroPython versions/ports/forks provide a different message after
# a Soft Reset, check if we are in raw REPL, if not send a CTRL-A again
data = serial.read_until(b'raw REPL; CTRL-B to exit\r\n>')
if not data.endswith(b'raw REPL; CTRL-B to exit\r\n>'):
    serial.write(b'\r\x01')
    data = serial.read_until(b'raw REPL; CTRL-B to exit\r\n>')
    if not data.endswith(b'raw REPL; CTRL-B to exit\r\n>'):
        if COMMAND_LINE_FLAG:
            print(data)
        raise IOError('Could not enter raw REPL.')
```